packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -59,6 +81,124 @@ class FramesBarChart extends CoreElement {
       _selectedFrameController.add(selectedFrame?.frame);
     }
   }
+*/
+}
+
+class PlotlyDivGraph extends CoreElement {
+  bool _processDatum;
+  Timer timer;
+  final FramesBarChart framesBarChart;
+  final TimelineFrame frame;
+  final Map<int, TimelineFrame> _frames = {};
+
+  ReadWriteMutex mutex = ReadWriteMutex();
+
+  List<int> dataIndexes = [];
+  List<num> cpuDurations = [];
+  List<num> gpuDurations = [];
+
+  FramesBarPlotly plotlyChart;
+
+  PlotlyDivGraph(this.framesBarChart, this.frame, [bool datum = true])
+      : this._processDatum = datum,
+        super('div') {
+    element.id = 'graph';
```

Comment:
is setting the id needed?
it is nice to avoid id whenever possible as there should only be one element matching an id in the document.

Reply:
id is how plotly finds the div to plot into (could use the Node but its so nice to work with the id).  I could construct an id that is unique at first usage and use that everywhere.  e.g.,

'GRAPH_${DateTime.now().millisecondsSinceEpoch}'

Maybe that's overkill?

Reply:
leaving the id is fine for now. Maybe just give it an id that is specific to the timeline view so we don't have a bug when another page of the app adds a chart?